src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int ARRAY_EXTEND_COEFFICIENT = 2;
+
+    private int arrayCapacity = DEFAULT_CAPACITY;
+    private int threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    private int size;
+    private Node[] hashMapArray = new Node[arrayCapacity];
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int elementPosition = findElementPosition(key);
+        if (hashMapArray[elementPosition] == null) {
+            hashMapArray[elementPosition] = new Node(key, value, null);
+            size++;
+            return;
+        }
+        Node node = hashMapArray[elementPosition];
+        while (true) {
+            if (key == null || node.key == null) {
+                if (key == null && node.key == null) {
+                    node.value = value;
+                    return;
+                }
+            } else if (node.key.equals(key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node(key, value, null);
+                size++;
+                return;
+            }
+            node = node.next;
```

Comment:
1. There is standard way of make null-safe comparison. If both are nulls then key == node.key will return true, if at least one is not null then we checking the one on which we will make call of equals() method to avoid NPE and safely calling equals method.
2. We should avoid using while(true) where possible
```suggestion
        while (node.next != null) {
            if (key == node.key || key != null && key.equals(node.key)) {
                node.value = value;
                return;
            }
            node = node.next;
        }
        node.next = new Node(key, value, null);
        size++;
        return;           
```

Reply:
I can't replace condition statement in while because, if next body is null, we haven't checked key of current node, so I've left the true condition statement

Reply:
You can replace it with this condition `while (node != null)`
We should avoid using `while(true) `